src/components/EditorWidgets/Object/ObjectPreview.js

```diff
@@ -1,12 +1,116 @@
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React from 'react';
+import { List } from 'immutable';
+import MarkdownPreview from '../Markdown/MarkdownPreview';
+import ImagePreview from '../Image/ImagePreview';
+import FilePreview from '../File/FilePreview';
+import RelationPreview from '../Relation/RelationPreview';
 
-const ObjectPreview = ({ field }) => (
-  <div className="nc-widgetPreview">{(field && field.get('fields')) || null}</div>
+const renderObject = (field, value, result, getAsset) => {
+  const fields = value.get('fields').toArray();
+  const resultToJS = result.toJS();
+  if (resultToJS) {
+    const values = List(
+      fields.map((childField) => {
+        const key = childField.get('name');
+        let renderedField = childField.merge({ [key]: result.get(key) || '' });
+        if (key === 'name') {
+          renderedField = renderedField.merge({ ___name: result.get(key) });
+        }
+        return renderedField || null;
+      })
+    );
+    return <ObjectPreview field={field} value={values} getAsset={getAsset} nested />;
+  }
+  return null;
+};
+
+const renderValue = (field, value, getAsset) => {
+  const widget = value.get('widget');
+  let result = value.get('___name') || value.get(value.get('name'));
+  let valuePreview = null;
+
+  switch (widget) {
```

Comment:
This switch statement ties this widget to the widgets below in an undesirable way - the object widget shouldn't know about these other widgets.